test/e2e/federated-service.go

```diff
@@ -168,8 +170,25 @@ var _ = framework.KubeDescribe("[Feature:Federation]", func() {
 				}
 
 				createBackendPodsOrFail(clusters, nsName, FederatedServicePodName)
+
 				service = createServiceOrFail(f.FederationClientset_1_5, nsName, FederatedServiceName)
-				waitForServiceShardsOrFail(nsName, service, clusters)
+				obj, err := conversion.NewCloner().DeepCopy(service)
+				// Cloning shouldn't fail. On the off-chance it does, we should
+				// shallow copy service to serviceShard before failing.
+				if err != nil {
```

Comment:
The test will fail, see `framework.ExpectNoError()` in the if block.

Also the problem with just failing the test without doing this shallow copy is, we will never really get a chance to clean up the underlying services when the cloner fails for a reason not in our control, such as cloner bug or such. That will cause resources to leak and the entire suite to fail soon.